### PR TITLE
Bug 1874048 - Add score to AMP suggestions

### PR DESCRIPTION
**Pull request by tiftran**
We added score to AMP suggestions in https://github.com/mozilla/application-services/pull/6055 so making changes here to accommodate it, I believe the application services PR needs to be merged before this one, so putting this as draft for now

### Pull Request checklist

- [ ] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Changelog**: This PR includes [a changelog entry](https://github.com/mozilla-mobile/firefox-android/blob/main/docs/changelog.md) or does not need one
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/firefox-android/blob/main/docs/shared/android/accessibility_guide.md) or does not include any user facing features

### After merge
- **Breaking Changes**: If this is a breaking Android Components change, please push a draft PR on [Reference Browser](https://github.com/mozilla-mobile/reference-browser) to address the breaking issues.

### To download an APK when reviewing a PR (after all CI tasks finished running):
1. Click on `Checks` at the top of the PR page.
2. Click on the `firefoxci-taskcluster` group on the left to expand all tasks.
3. Click on the `build-apk-{fenix,focus,klar}-debug` task you're interested in.
4. Click on `View task in Taskcluster` in the new `DETAILS` section.
5. The APK links should be on the right side of the screen, named for each CPU architecture.

### GitHub Automation
https://bugzilla.mozilla.org/show_bug.cgi?id=1874048

### CODE REVIEW

File: android-components/components/feature/fxsuggest/src/main/java/mozilla/components/feature/fxsuggest/FxSuggestSuggestionProvider.kt
Comment (linabutler) [inline android-components/components/feature/fxsuggest/src/main/java/mozilla/components/feature/fxsuggest/FxSuggestSuggestionProvider.kt:147]: Hmm, `SuggestionDetails` is an internal class, and we don't use `details.score` anywhere—I don't think we need to add this just yet.

**Review by linabutler — APPROVED** [review]

File: android-components/components/feature/fxsuggest/src/test/java/mozilla/components/feature/fxsuggest/FxSuggestSuggestionProviderTest.kt
Comment (linabutler) [inline android-components/components/feature/fxsuggest/src/test/java/mozilla/components/feature/fxsuggest/FxSuggestSuggestionProviderTest.kt:101]: (But we do need these additions for sure—thanks!)

File: android-components/components/feature/fxsuggest/src/main/java/mozilla/components/feature/fxsuggest/FxSuggestSuggestionProvider.kt
Comment (tiftran) [reply android-components/components/feature/fxsuggest/src/main/java/mozilla/components/feature/fxsuggest/FxSuggestSuggestionProvider.kt:147]: kk

**Comment by MatthewTighe** [comment]
Are there more changes to come, or will this be the complete diff? If the latter, no need to wait for an additional review from me once the A-S change lands

**Comment by mcarare** [comment]
Commit fix cherry-picked to https://github.com/mozilla-mobile/firefox-android/pull/5288 as they cannot land separately.

**Pull request closed without merging**
